src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -10,7 +10,26 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
-        public static void GetByteCount(Encoding encoding, string chars, int index, int count, int expected)
+        public static void Encode(Encoding encoding, string chars, int index, int count, byte[] expected)
+        {
+            byte[] fullArray = new byte[expected.Length + 4];
+            for (int i = 0; i < fullArray.Length; i++)
```

Comment:
Well @stephentoub suggested that we have a test to make sure other stuff in the array outside the range of the byte index and count are unaffected, this does so